autoload/go/tool.vim

```diff
@@ -162,6 +162,18 @@ function! go#tool#FilterValids(items) abort
 endfunction
 
 function! go#tool#ExecuteInDir(cmd) abort
+  " Verify that the directory actually exists. If the directory does not
+  " exist, then assume that the a:cmd should not be executed. Callers expect
+  " to check v:shell_error (via go#util.ShellError()), so execute a command
+  " that will return an error as if a:cmd was run and exited with an error.
+  " This helps avoid errors when working with plugins that use virtual files
+  " that don't actually exist on the file system (e.g. vim-fugitive's
+  " GitDiff).
+  if !isdirectory(expand("%:p:h"))
+    let out = go#util#System("exit 1")
```

Comment:
Could you use `false`, instead of `exit`? The former is in POSIX, the latter isn't. Also `exit` is usually a shell built-in, so this code depends on the shell.

You should also use `go#util#Exec()` for all new code, the `System()` function will be removed once all calls are rewritten to use that.